### ISSUE TITLE
Cognito - Cannot map phone_number to phoneNumbers Google attribute

### ISSUE THREAD

**Issue by MensurRasic** [aws-amplify/amplify-js#11623]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

Authentication

### Amplify Categories

auth

### Environment information

<details>

```
System:
    OS: macOS 13.4
    CPU: (10) arm64 Apple M1 Pro
    Memory: 41.39 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.2.0 - /opt/homebrew/bin/node
    Watchman: 2023.05.15.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 114.0.5735.198
    Safari: 16.5
  npmPackages:
    @aws-amplify/analytics: ^6.0.16 => 6.3.2 
    @aws-amplify/rtn-push-notification: ^1.1.1 => 1.1.1 
    @babel/core: ^7.12.9 => 7.22.8 
    @babel/runtime: ^7.12.5 => 7.22.6 
    @react-native-async-storage/async-storage: ^1.19.0 => 1.19.0 
    @react-native-community/eslint-config: ^2.0.0 => 2.0.0 
    @react-native-community/netinfo: ^8.3.1 => 8.3.1 
    @react-native-community/push-notification-ios: ^1.10.1 => 1.11.0 
    @react-native-picker/picker: ^2.4.4 => 2.4.10 
    @react-native/metro-config: ^0.72.7 => 0.72.7 
    @react-navigation/bottom-tabs: ^6.3.2 => 6.5.8 
    @react-navigation/material-top-tabs: ^6.6.2 => 6.6.3 
    @react-navigation/native: ^6.0.11 => 6.1.7 
    @react-navigation/native-stack: ^6.7.0 => 6.9.13 
    HelloWorld:  0.0.1 
    amazon-cognito-identity-js: ^5.2.14 => 5.2.14 (6.3.1)
    amazon-cognito-identity-js/internals:  undefined ()
    aws-amplify: ^5.3.3 => 5.3.3 
    aws-amplify-react-native: ^7.0.2 => 7.0.2 
    aws-sdk: ^2.1220.0 => 2.1413.0 
    eslint: ^7.32.0 => 7.32.0 
    i: ^0.3.7 => 0.3.7 
    i18next: ^21.8.14 => 21.10.0 
    jest: ^26.6.3 => 26.6.3 
    metro-react-native-babel-preset: ^0.70.3 => 0.70.4 (0.76.5)
    npm: ^9.2.0 => 9.8.0 
    react: 18.2.0 => 18.2.0 
    react-devtools: 4.25.0 => 4.25.0 
    react-devtools-core: 4.25.0 => 4.25.0 (4.27.8)
    react-native: 0.72.1 => 0.72.1 
    react-native-get-random-values: ^1.9.0 => 1.9.0 
    react-native-reanimated: ^3.3.0 => 3.3.0 
  npmGlobalPackages:
    @aws-amplify/cli: 12.1.1
    ios-deploy: 1.12.1
    jest: 29.0.3
    jetifier: 2.0.0
    npm: 9.6.6
    react-devtools-core: 4.25.0
    react-devtools: 4.27.1
    react-native-cli: 2.0.1
    react-native-rename: 2.9.0
    react-native: 0.72.1


```

</details>

### Describe the bug

I am trying to setup a [social sign-in (Oauth) with google](https://docs.amplify.aws/lib/auth/social/q/platform/react-native/#setup-your-auth-provider) in my react-native app.
All the setup works fine until I need to map attributes between google and my user pool.

Here is the OpenID connect scopes from my app client web : 
<img width="820" alt="Screenshot 2023-07-13 at 4 46 10 PM" src="https://github.com/aws-amplify/amplify-js/assets/23143508/10511590-5ad6-4a35-85fa-5515212d778c">

As mentioned below in my aws-export.js, phone and email are both in the oath object.

When I try to sign-in with a google account, that actually has, or not, a phone number, after selecting my google accountI always get this error : 
`Sign in failure {"channel": "auth", "patternInfo": [], "payload": {"data": [Error: attributes+required%3A+%5Bphone_number%5D], "event": "signIn_failure", "message": "The OAuth response flow failed"}, "source": "Auth"}`

From my comprehension the mapping is properly set in `Cognito` as well : 
<img width="831" alt="Screenshot 2023-07-13 at 4 49 44 PM" src="https://github.com/aws-amplify/amplify-js/assets/23143508/e1a59f90-2e60-473b-beda-2572dfa2c0e2">

I also tried to hardcode the google attribute into a string or any other possible value such as `+12223334444`, but it still fail.

I don't receive any error for the other fields.

Any temporary solution/fix for this? 

### Expected behavior

The user should be able to sign-in with google account.

### Reproduction steps

1. Implement the social sign-in (google) in a react-native app (0.72.1)
2. Sign-in with a google account that has a name, phone number and email

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### aws-exports.js

const awsmobile = {
    "aws_project_region": "us-west-1",
    "aws_cognito_identity_pool_id": "us-west-1:xxx",
    "aws_cognito_region": "us-west-1",
    "aws_user_pools_id": "us-west-1_xxx",
    "aws_user_pools_web_client_id": "xxx",
    "oauth": {
        "domain": "xxx.auth.us-west-1.amazoncognito.com",
        "scope": [
            "phone",
            "email",
            "openid",
            "profile",
            "aws.cognito.signin.user.admin"
        ],
        "redirectSignIn": "xxx://",
        "redirectSignOut": "xxx://",
        "responseType": "code"
    },
    "federationTarget": "COGNITO_USER_POOLS",
    "aws_cognito_username_attributes": [],
    "aws_cognito_social_providers": [
        "GOOGLE"
    ],
    "aws_cognito_signup_attributes": [
        "EMAIL"
    ],
    "aws_cognito_mfa_configuration": "OFF",
    "aws_cognito_mfa_types": [
        "SMS"
    ],
    "aws_cognito_password_protection_settings": {
        "passwordPolicyMinLength": 8,
        "passwordPolicyCharacters": []
    },
    "aws_cognito_verification_mechanisms": [
        "EMAIL"
    ]
};

### Mobile Device

iPhone X, Samsung A32

**Comment by cwomack** [comment]
Hello, @MensurRasic 👋.  Can check to see if that `phone_number` attribute is currently showing under the "Required attributes" for the User Pool of your app? I don't think it will be based on your `aws-exports.js` having only "EMAIL" within the "aws_cognito_signup_attributes" array, but want to be sure. You can find this within the [Cognito console](https://console.aws.amazon.com/cognito/home) > [user pool name] > Sign-up experience tab.  

Providing it's not, would you see if _removing_ the mapped attribute of `phone_number: phoneNumbers` resolves the issue? This can be done under the "Sign-in experience" tab where you set up the Federated identity provider.

**Comment by MensurRasic** [comment]
Hey, yes phone_number is a required attribute.
<img width="1054" alt="Screenshot 2023-07-13 at 6 02 30 PM" src="https://github.com/aws-amplify/amplify-js/assets/23143508/abdb222d-f6cd-4ad5-a1a3-6891f0b61a65">

I can remove it from `OpenID Connect scopes` but I have to provide a value in the attribute mapping : 
<img width="824" alt="Screenshot 2023-07-13 at 6 09 24 PM" src="https://github.com/aws-amplify/amplify-js/assets/23143508/67b35f95-0a7b-40e1-96c8-4dcc0782b3fa">

I am not sure if I can pass there a hard coded value and force the user to update it later within the app. 
I tried all the [possible format](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html?icmpid=docs_cognito_console_help_panel) none of them works.

On my side, I am suspecting that I need to provide more [reviews and requirements from google to add additional scopes](https://developers.google.com/workspace/guides/configure-oauth-consent#choose-scopes) because I don't see the phone number as a shared scope in the google sign-in screen : 
![IMG_38E62C77E02B-1](https://github.com/aws-amplify/amplify-js/assets/23143508/e374cb12-7896-4f06-a7b7-922198962484)

**Comment by MensurRasic** [comment]
I managed to add additional scopes in the `OAuth consent screen`, it's counter intuitive if you only read the [documentation (step 8)](https://docs.amplify.aws/lib/auth/social/q/platform/react-native/#setup-your-auth-provider) : 
<img width="602" alt="Screenshot 2023-07-13 at 9 15 47 PM" src="https://github.com/aws-amplify/amplify-js/assets/23143508/bef170a0-a62c-43f6-b199-21bf85427270">

But it wasn't enough, I realized I forgot to add `phone` in the `Authorized scopes` :
<img width="1390" alt="Screenshot 2023-07-13 at 9 18 43 PM" src="https://github.com/aws-amplify/amplify-js/assets/23143508/3d78169b-53a6-4138-9432-a49a3a491c41">

Adding the `phone` authorization scope has no effect on the OAuth consent screen.
----
On the other hand, I found this [thread](https://community.auth0.com/t/phone-number-not-returned-in-user-profile-from-google-oauth/48357) that might be insightful.
**TLDR** : _In our Basic and Extended profiles we do not provide phone number, if you hover over the
Basic and Extended profiles it mentions the fields listed and does not include phoneNumber._ 

If we are not able to get the phoneNumber from OAuth when it's mandatory in Cognito, we should at least be able to pass a temporary value, let me know if you have anything on your side!

**Comment by cwomack** [comment]
@MensurRasic, appreciate your patience as I've been trying to test this to determine the next best steps.  Google does **_not_** expose phone numbers as part of [their OAuth scopes](https://www.googleapis.com/discovery/v1/apis/oauth2/v2/rest?fields=auth(oauth2(scopes))).  This means that if `phone_number` has been set as a required attribute within the Cognito User Pool, the federated sign-in process will continue to fail while attempting to retrieve the attribute from Google's OAuth response.  

One way to resolve the issue would be to create a new user pool for your app and make sure that `phone_number` is not a required attribute.  I've been trying to see if there's a workaround that doesn't require you to do this, but if your Cognito User Pool isn't too large yet and the app is in the early stages... you'd save a lot of headache by creating a new one.  

Is this something that you'd be willing to do?

**Comment by cwomack** [comment]
@MensurRasic, just wanted to follow up and let you know that I'll be creating a callout within our docs on the [Social Sign-in](https://docs.amplify.aws/lib/auth/social/q/platform/js/) page to warn future users of the limitation/hurdle you ran into.  I know this doesn't resolve the issue you're experiencing, but hopefully we can prevent others from running into the same problem.  Apologies you ran into this, but let me know if there's anything else I can do to assist.

**Comment by dovedevic** [comment]
Writing in to add to this, I have an old cognito pool that uses birthday as a required attribute. Because of this requirement, Apple SSO is not possible for similar reasons above. While I think having a warning is a great step in the right direction for warning on pool creation that "X required attribute will restrict you from using Y popular SSO", I do think at some point in product maturity we should be able to *remove* required attributes. I understand restricting required attribute changes when *new* ones are added, but I feel removing should at least be an option, or if not, better / more seamless pool migration routines should exist in it's place.

**Comment by cwomack** [comment]
@dovedevic, appreciate you adding this comment after running into a similar issue with OIDC scopes (Sign in with Apple in your case) and bringing it to our Discord Office hours.  I'll also review this issue internally with our team to see if there's potentially something more beyond a docs update that can be done.  At this time, you unfortunately cannot update/alter required attributes on a Cognito user pool once it's been created.

**Comment by cwomack** [comment]
Closing this issue now that the documentation improvements have been made.  Sorry you had to experience this @MensurRasic, but hoping [this update](https://docs.amplify.aws/lib/auth/social/q/platform/js/#:~:text=When%20configuring%20social,new%20user%20pool.) makes it more clear for future readers trying to implement Social Sign-in.
